File: C++/8 kyu - Convert number to reversed array of digits.cpp
// https://www.codewars.com/kata/convert-number-to-reversed-array-of-digits/train/cpp
// My solution
std::vector<int> digitize(unsigned long n) {
    std::vector<int> digits;
    while(n > 0) {
        digits.push_back(n % 10);
        n = n / 10;
    }
    return digits;
}

// ...
std::vector<int> digitize(unsigned long n) {        
    std::vector<int> rod;
    do {
        rod.push_back(n % 10);
    } while (n /= 10);
    return rod;
}

// ...
#include <vector>
#include <string>
using namespace std;

vector<int> digitize(unsigned long n) {
    vector<int> result;
    string digits = to_string(n);
    for (int i = digits.length()-1; i >= 0; i--) {
        int item = digits[i]-'0';
        result.push_back(item);
    }
    return result;
}

// ...

File: Java/8 kyu - Convert number to reversed array of digits.java
// https://www.codewars.com/kata/convert-number-to-reversed-array-of-digits/train/java
// My solution
public class Kata {
    public static int[] digitize(long n) {
        String string = String.valueOf(n);
        int[] array = new int[string.length()];
        int strLength = string.length();
        for(int i = strLength; i > 0 ; i--)
            array[strLength - i] = Character.getNumericValue(string.charAt(i - 1));
        return array;
    }
}

// ...
public class Kata {
    public static int[] digitize(long n) {
        return new StringBuilder().append(n)
                                .reverse()
                                .chars()
                                .map(Character::getNumericValue)
                                .toArray();
    }
}

// ...
import java.lang.Math;
public class Kata {
    public static int[] digitize(long n) {
        String s = String.valueOf(n);
        int length = s.length();
        int[] array = new int[length];
        for ( int i = 0; i < length; i++)
            array[i] = (int) (s.charAt(length - i - 1)) - 48;
        return array;
    }
}

// ...
import java.util.Arrays;
import java.util.Stack;

public class Kata {
    public static int[] digitize(long n) {
        Stack<Integer> digits = new Stack<>();

        while (n > 0) {
            digits.push((int) (n % 10));
            n /= 10;
        }

        return digits.stream().mapToInt(Integer::intValue).toArray();
    }
}

// ...
public class Kata {
    public static int[] digitize(long n) {
        char[] array = String.valueOf(n).toCharArray();
        int[] result = new int[array.length];
        
        for(int i =0; i < array.length;i++)
            result[array.length-1-i] = Integer.valueOf(String.valueOf(array[i]));
        
        return result;
    }
}